File: 07-SimpleSymbols/7.hs
--Given a string with alphabet [A-z],+,=: Does every letter have two "+"'s around it?
simpleSymbols :: String -> Bool
simpleSymbols "" = True
simpleSymbols xs@('+':x:'+':rest)
  | elem x "=+" = simpleSymbols $ tail xs
  | otherwise   = simpleSymbols ('+':rest)
simpleSymbols (x:xs) 
  | elem x "=+" = simpleSymbols xs
  | otherwise   = False

main = do
  print $ simpleSymbols "+a++b+===+" == True
  print $ simpleSymbols "++abc++"    == False

File: 07-SimpleSymbols/7.js
function SimpleSymbols(str) { 
  for(var i=0; i < str.length; i++){
    if((/[a-z]/gi.test(str[i])) && (i === 0 || i === str.length-1 || (str[i-1] !== str[i+1]) || str[i-1] !== "+")){
        return false
    }
  }
  
  return true;
}

function SimpleSymbolsRegex(str){
  return !/(^|[^+])[a-z]|[a-z]([^+]|$)/gi.test(str)
}

console.log(SimpleSymbols("+a++b+===+") === true);